internal/server/open_linux.go

```diff
@@ -0,0 +1,6 @@
+//go:build linux
+// +build linux
+
+package server
+
+const openCommand = "xdg-open"
```

Comment:
Minor, and likely not worth it, but I could see this being a separate package that encapsulates the OS build logic and `exec.Command` logic.

Reply:
Ooh, say more?

Reply:
It's not a fully formed thought yet, but I was thinking something like `internal/oscommand`:

```go
//go:build linux

func Open(target string) error {
    cmd := exec.Command("xdg-open", target)
    return cmd.Run()
}
```

and 

```
//go:build darwin

func Open(target string) error {
    cmd := exec.Command("open", target)
    return cmd.Run()
}
```

I was also wondering if it would make sense to consolidate the clipboard logic here too, maybe implementing some kind of `SystemCommand` interface:

I was roughly thinking like:

```
type SystemCommands interface {
  Copy(string) error
  Paste() ([]byte, error)
  Open() error
}
```

I imagine the big benefit there would be that we could consolidate all of our OS specific logic in a single place instead of scattering it throughout several different files. It might be easier to reason about and modify, but definitely open to hearing thoughts on that idea, for or against.

Reply:
@BlakeWilliams WDYT about 47d1953b2ace26f89856f08ab12a707fa53045c9?

Reply:
I had some questions around the use of struct embedding since I've mostly seen it used to simulate inheritance, but overall LGTM